website/docs/advanced-topics/immutable.mdx

```diff
@@ -0,0 +1,84 @@
+---
+title: 'Immutable'
+description: 'Immutable.js-like types for Yew'
+---
+
+## What are immutable types?
+
+These are types that you can instantiate but never mutate the values. In order
+to update a value, you must instantiate a new value.
+
+## Why using immutable types?
+
+In a React-like framework, properties are propagated from ancestors to
+children. This means that the properties must live when each component is
+updated. This is why properties should —ideally— be cheap to clone. In order to
+achieve this we usually wrap things in `Rc`.
+
+Immutable types are a great fit for holding property's values because they can
+be cheaply cloned when passed from component to component.
+
+## Example using an immutable array
+
+```rust
+use wasm_bindgen::JsCast;
+use wasm_bindgen::UnwrapThrowExt;
+use web_sys::{HtmlInputElement, KeyboardEvent};
+use yew::immutable::*;
+use yew::prelude::*;
+
+#[derive(Properties, PartialEq)]
+struct FolksViewProps {
+    folks: IArray<IString>,
+}
+
+#[function_component(FolksView)]
+fn folks_view(props: &FolksViewProps) -> Html {
+    html! {
+        <>
+        <p>{"Hello to:"}</p>
+        <ul>
+        { for props.folks.iter().map(|s| html!(<li>{s}</li>)) }
+        </ul>
+        </>
+    }
+}
+
+#[function_component(ArrayExample)]
+pub fn array_example() -> Html {
+    let folks = use_state(IArray::<IString>::default);
+    let onkeyup = {
+        let folks = folks.clone();
+        Callback::from(move |e: KeyboardEvent| {
+            if e.key() == "Enter" {
+                let event: Event = e.dyn_into().unwrap_throw();
+                let event_target = event.target().unwrap_throw();
+                let target: HtmlInputElement = event_target.dyn_into().unwrap_throw();
+                let name = target.value();
+                target.set_value("");
+
+                folks.set(
+                    folks
+                        .iter()
+                        .chain(std::iter::once(IString::from(name)))
+                        .collect(),
+                );
+            }
+        })
+    };
+
+    html! {
+        <>
+        <h2>{"Input"}</h2>
+        <input {onkeyup} />
+        <h2>{"Output"}</h2>
+        <FolksView folks={&*folks} />
+        </>
+    }
+}
+```
```

Comment:
There's no need to show all that code in a documentation page. A complete example should be in the examples directory

Reply:
Good point